frontends/web/src/components/amount/amount.test.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { render } from '@testing-library/react';
+import { Amount } from './amount';
+
+describe('Amount formatting', () => {
+  let coins;
+
+  describe('sat amounts', () => {
+    coins = ['sat', 'tsat'];
+    coins.forEach(coin => {
+      it('12345678901234 ' + coin + ' with removeBtcTrailingZeroes enabled gets spaced', () => {
+        const { getByText } = render(<Amount amount="12345678901234" unit={coin} removeBtcTrailingZeroes/>);
+        expect(getByText('12')).toBeInTheDocument();
```

Comment:
i removed the `.reverse()` in the code and all unit tests still passed. not sure if it is worth it to improve the tests - i checked that it is correct manually - but such a bug would be devestating.

Reply:
Good catch! I updated the test to actually check for the HTML and detect this kind of error